libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -2033,3 +2033,24 @@ string YulUtilFunctions::revertReasonIfDebug(string const& _message)
 {
 	return revertReasonIfDebug(m_revertStrings, _message);
 }
+
+string YulUtilFunctions::returnDataToArrayFunction()
+{
+	return m_functionCollector->createFunction("returnDataToArrayFn", [&]() {
+		return !m_evmVersion.supportsReturndata()
+			? "return "s + to_string(CompilerUtils::zeroPointer)
+			: R"(
+				let v := returndatasize()
+				switch v
+				case 0 {
+					v := 0x60
+				}
+				default {
+					v := mload(0x40)
```

Comment:
Please use `allocateMemoryArrayFunction`

Reply:
Note to myself:  parameter is ArrayType, hence: `TypeProvider::bytesMemory()` might be of help here. (TODO)